.github/workflows/vet.yml

```diff
@@ -9,14 +9,17 @@ on:
 permissions:
   contents: read  #  to fetch code (actions/checkout)
 
+env:
+  GOTOOLCHAIN: local
+
 jobs:
   vet:
     runs-on: ubuntu-latest
     steps:
     - uses: actions/checkout@v4
     - uses: actions/setup-go@v5
       with:
-        go-version: '1.20.x'
+        go-version: '1.21.x'
     - name: Install tools
       run: |
         go install golang.org/x/lint/golint@latest && \
```

Comment:
Can you restore `go install honnef.co/go/tools/cmd/staticcheck@v0.4.7` to `go install honnef.co/go/tools/cmd/staticcheck@latest` on line 27, below? Thanks!

Reply:
The latest staticcheck requires 1.22: https://github.com/dominikh/go-tools/blob/56172d41b117cc2c2f99f65fe0a790c8d7d7ea66/go.mod#L3

One thing we could consider is only running vet on latest version. We do this in other repos today.

Reply:
Until now, I thought it was good to run vet on the oldest version in order to keep `toolchain` from sneaking into go.mod files in our repo. But maybe we no longer have any oldest version concerns like that.

Reply:
> The latest staticcheck requires 1.22

See you in 2025, staticcheck!